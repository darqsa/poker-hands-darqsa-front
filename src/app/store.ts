import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import handsSlice from "../features/hands/slices/handsSlice";
import uiSlice from "../features/ui/slices/uiSlice";
import userSlice from "../features/users/slices/userSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    hands: handsSlice,
    ui: uiSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
