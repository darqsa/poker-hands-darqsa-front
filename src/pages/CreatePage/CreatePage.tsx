import HandForm from "../../components/HandForm/HandForm";

const CreatePage = (): JSX.Element => {
  return <HandForm formFunction="create" />;
};
export default CreatePage;
