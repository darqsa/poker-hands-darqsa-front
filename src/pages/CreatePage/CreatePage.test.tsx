import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../app/store";
import CreatePage from "./CreatePage";

describe("Given a create page component", () => {
  describe("When invoked", () => {
    test("Then it should match the snapshot", () => {
      const expectedCreatePage = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <CreatePage />
          </BrowserRouter>
        </Provider>
      );

      expect(expectedCreatePage).toMatchSnapshot();
    });
  });
});
