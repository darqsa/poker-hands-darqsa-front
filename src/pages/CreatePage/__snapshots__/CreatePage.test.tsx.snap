// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a create page component When invoked Then it should match the snapshot 1`] = `
<form
  className="sc-gsnTZi sc-dkzDqf ehTXBa jbBMgv form"
  onSubmit={[Function]}
>
  <h2
    className="form__title"
  >
    Game info
  </h2>
  <section
    className="form__section"
  >
    <div
      className="form__players-container"
    >
      <div
        className="form__player-container"
      >
        <h3
          className="form__group-heading"
        >
          Hero
        </h3>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="heroPosition"
          >
            * Position
          </label>
          <select
            autoComplete="off"
            className="form__input form__input--selector"
            id="heroPosition"
            onChange={[Function]}
            required={true}
            value={0}
          >
            <option
              className="form__list-option"
              value={0}
            >
              ---
            </option>
            <option
              className="form__list-option"
              value={1}
            >
              SB
            </option>
            <option
              className="form__list-option"
              value={2}
            >
              BB
            </option>
            <option
              className="form__list-option"
              value={3}
            >
              UTG
            </option>
            <option
              className="form__list-option"
              value={4}
            >
              MP
            </option>
            <option
              className="form__list-option"
              value={5}
            >
              CO
            </option>
            <option
              className="form__list-option"
              value={6}
            >
              BTN
            </option>
          </select>
        </div>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="heroStack"
          >
            * Stack
          </label>
          <input
            autoComplete="off"
            className="form__input form__input--number"
            id="heroStack"
            max={999}
            onChange={[Function]}
            required={true}
            type="number"
            value=""
          />
        </div>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="heroCard1"
          >
            * Hand
          </label>
          <div
            className="form__hand-container"
          >
            <input
              autoComplete="off"
              className="form__input form__input--hand"
              id="heroCard1"
              maxLength={2}
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <input
              autoComplete="off"
              className="form__input form__input--hand"
              data-testid="hero-hand-2"
              id="heroCard2"
              maxLength={2}
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="form__player-container"
      >
        <h3
          className="form__group-heading"
        >
          Villain
        </h3>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="villainPosition"
          >
            * Position
          </label>
          <select
            autoComplete="off"
            className="form__input form__input--selector"
            id="villainPosition"
            onChange={[Function]}
            required={true}
            value={0}
          >
            <option
              className="form__list-option"
              value={0}
            >
              ---
            </option>
            <option
              className="form__list-option"
              value={1}
            >
              SB
            </option>
            <option
              className="form__list-option"
              value={2}
            >
              BB
            </option>
            <option
              className="form__list-option"
              value={3}
            >
              UTG
            </option>
            <option
              className="form__list-option"
              value={4}
            >
              MP
            </option>
            <option
              className="form__list-option"
              value={5}
            >
              CO
            </option>
            <option
              className="form__list-option"
              value={6}
            >
              BTN
            </option>
          </select>
        </div>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="villainStack"
          >
            * Stack
          </label>
          <input
            autoComplete="off"
            className="form__input form__input--number"
            id="villainStack"
            max={999}
            onChange={[Function]}
            required={true}
            type="number"
            value=""
          />
        </div>
        <div
          className="form__group"
        >
          <label
            className="form__label"
            htmlFor="villainCard1"
          >
            * Hand
          </label>
          <div
            className="form__hand-container"
          >
            <input
              autoComplete="off"
              className="form__input form__input--hand"
              id="villainCard1"
              maxLength={2}
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <input
              autoComplete="off"
              className="form__input form__input--hand"
              data-testid="villain-hand-2"
              id="villainCard2"
              maxLength={2}
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="handName"
      >
        * Hand name
      </label>
      <input
        autoComplete="off"
        className="form__input form__input--text"
        id="handName"
        maxLength={40}
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium form__icon css-i4bv87-MuiSvgIcon-root"
      data-testid="next-first-page"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </section>
</form>
`;
